src/avx2/deser.rs

```diff
@@ -17,192 +17,190 @@ use crate::stringparse::{handle_unicode_codepoint, ESCAPE_MAP};
-        // create sub sliced form sub to `sub.len()`.
-
-        let src: &[u8] = unsafe { data.get_kinda_unchecked(idx..) };
-        let mut src_i: usize = 0;
-        let mut len = src_i;
-        loop {
-            let v: __m256i = unsafe {
-                _mm256_loadu_si256(src.as_ptr().add(src_i).cast::<std::arch::x86_64::__m256i>())
-            };
-
-            // store to dest unconditionally - we can overwrite the bits we don't like
-            // later
-            let bs_bits: u32 = unsafe {
-                static_cast_u32!(_mm256_movemask_epi8(_mm256_cmpeq_epi8(
-                    v,
-                    _mm256_set1_epi8(b'\\' as i8)
-                )))
-            };
-            let quote_mask = unsafe { _mm256_cmpeq_epi8(v, _mm256_set1_epi8(b'"' as i8)) };
-            let quote_bits = unsafe { static_cast_u32!(_mm256_movemask_epi8(quote_mask)) };
-            if (bs_bits.wrapping_sub(1) & quote_bits) != 0 {
-                // we encountered quotes first. Move dst to point to quotes and exit
-                // find out where the quote is...
-                let quote_dist: u32 = quote_bits.trailing_zeros();
-
-                ///////////////////////
-                // Above, check for overflow in case someone has a crazy string (>=4GB?)
-                // But only add the overflow check when the document itself exceeds 4GB
-                // Currently unneeded because we refuse to parse docs larger or equal to 4GB.
-                ////////////////////////
-
-                // we advance the point, accounting for the fact that we have a NULl termination
-
-                len += quote_dist as usize;
-                unsafe {
-                    let v = std::str::from_utf8_unchecked(std::slice::from_raw_parts(
-                        input.add(idx),
-                        len,
-                    ));
-                    return Ok(v);
-                }
-
-                // we compare the pointers since we care if they are 'at the same spot'
-                // not if they are the same value
-            }
-            if (quote_bits.wrapping_sub(1) & bs_bits) == 0 {
-                // they are the same. Since they can't co-occur, it means we encountered
-                // neither.
-                src_i += 32;
-                len += 32;
-            } else {
-                // Move to the 'bad' character
-                let bs_dist: u32 = bs_bits.trailing_zeros();
-                len += bs_dist as usize;
-                src_i += bs_dist as usize;
-                break;
+#[target_feature(enable = "avx2")]
+#[allow(
+    clippy::if_not_else,
+    clippy::transmute_ptr_to_ptr,
+    clippy::too_many_lines,
+    clippy::cast_ptr_alignment,
+    clippy::cast_possible_wrap,
+    clippy::if_not_else,
+    clippy::too_many_lines
+)]
+#[cfg_attr(not(feature = "no-inline"), inline)]
+pub(crate) unsafe fn parse_str_avx<'invoke, 'de>(
```

Comment:
I'm not a fan of `unsafe` on the function level, since it makes proof-reading the code harder. I just added it since `#[target_feature(enable = "sse4.2")]` requires it

Reply:
fair point, I'm OK with that 👍